man/org.freedesktop.login1.xml

```diff
@@ -674,15 +679,20 @@ node /org/freedesktop/login1 {
       logs in or out, or a seat is added or removed. They each contain the ID of the object plus the object
       path.</para>
 
-      <para>The <function>PrepareForShutdown()</function> and <function>PrepareForSleep()</function> signals
-      are sent right before (with the argument <literal>true</literal>) or after (with the argument
+      <para>The <function>PrepareForShutdown()</function>,
+      <function>PrepareForShutdownWithMetadata()</function> and <function>PrepareForSleep()</function>
+      signals are sent right before (with the argument <literal>true</literal>) or after (with the argument
       <literal>false</literal>) the system goes down for reboot/poweroff and suspend/hibernate,
       respectively. This may be used by applications to save data on disk, release memory, or do other jobs
       that should be done shortly before shutdown/sleep, in conjunction with delay inhibitor locks. After
       completion of this work they should release their inhibition locks in order to not delay the operation
       any further. For more information see
-      <ulink url="https://www.freedesktop.org/wiki/Software/systemd/inhibit">Inhibitor Locks</ulink>.
-      </para>
+      <ulink url="https://www.freedesktop.org/wiki/Software/systemd/inhibit">Inhibitor Locks</ulink>. The
+      <function>PrepareForShutdownWithMetadata()</function> signal additionally sends a list of key/value
+      pair metadata fields. Currently it sends a <varname>start</varname> boolean (as defined above) and a
+      <varname>type</varname> string which defines the type of shutdown. The type can be one of
+      <literal>power-off</literal>, <literal>reboot</literal>, <literal>halt</literal>,
```

Comment:
I think this should be `poweroff` instead of `power-off`. `poweroff` is used as a value of various config. options and as a value of the `type` argument of `org.freedesktop.logind1.Manager.ScheduleShutdown()`. `power-off` is only used in polkit actions `org.freedesktop.login1.power-off*`.

Reply:
This is what gets logged as structure metadata - ie: `SHUTDOWN=power-off`, given it's structured that's also an API to communicate the shutdown type, so I think it makes more sense to make the signal metadata and the log metadata match, as they are more related

Reply:
() should not be used after a signal name, it is not a function call.

Reply:
other signals had it too - fixed in all of them